snuba/migrations/snuba_migrations/transactions/0016_transactions_add_group_ids_column.py

```diff
@@ -0,0 +1,86 @@
+from typing import List, Sequence, Tuple
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+new_columns: Sequence[Tuple[Column[Modifiers], str]] = [
+    (Column("group_ids", Array(UInt(64))), "timestamp"),
+    (
+        Column(
+            "_group_ids_hashed",
+            Array(
+                UInt(64),
+                Modifiers(materialized="arrayMap(t -> cityHash64(t), group_ids)"),
+            ),
+        ),
+        "group_ids",
+    ),
+]
+
+new_indexes: List[operations.SqlOperation] = [
+    operations.AddIndex(
+        storage_set=StorageSetKey.TRANSACTIONS,
+        table_name="transactions_local",
+        index_name="bf_group_ids_hashed",
+        index_expression="_group_ids_hashed",
+        index_type="bloom_filter()",
+        granularity=1,
+    )
```

Comment:
I'd suggest we started without index.
If the assumption that the vast majority of transactions will not have issues, then we are likely not going to gain much by adding an index (it may actually be counter productive as we would have to load the index as well during queries).

Specifically if the array is empty most of the time, I suspect clickhouse will be as fast at excluding those blocks where the column is always empty than it would be through the index.

Reply:
With the current detection logic we're finding that around 3-4 % of transaction events contain some kind of performance issue but with group creation rate limit and better thresholds, the actual rate will be even lower